strawberryfields/configuration.py

```diff
@@ -15,138 +15,229 @@
+
+    return parsed_config, filepath
+
+def load_config_file(filepath):
+    """Load a configuration object from a TOML formatted file.
+
+    Args:
+        filepath (str): path to the configuration file
+
+    Returns:
+         dict[str, dict[str, Union[str, bool, int]]]: the configuration
+            object that was loaded
+    """
+    with open(filepath, "r") as f:
+        config_from_file = toml.load(f)
+    return config_from_file
+
+def update_with_other_config(config, other_config):
+    """Updates the current configuration object with another one.
+
+    Args:
+        config (dict[str, dict[str, Union[str, bool, int]]]): the
+            configuration to be updated
+        other_config (dict[str, dict[str, Union[str, bool, int]]]): the
+            configuration used for updating
+
+    Returns:
+        dict[str, dict[str, Union[str, bool, int]]]): the updated
+            configuration
+    """
+    # Here an example for section is API
+    for section, sectionconfig in config.items():
+        for key in sectionconfig:
+            if key in other_config[section]:
+                config[section][key] = other_config[section][key]
+
+def update_from_environment_variables(config):
+    """Updates the current configuration object from data stored in environment
+    variables.
+
+    .. note::
+
+        The following environment variables are checked:
+
+        * SF_API_AUTHENTICATION_TOKEN
+        * SF_API_HOSTNAME
+        * SF_API_USE_SSL
+        * SF_API_DEBUG
+        * SF_API_PORT
+
+    Args:
+        config (dict[str, dict[str, Union[str, bool, int]]]): the
+            configuration to be updated
+    Returns:
+        dict[str, dict[str, Union[str, bool, int]]]): the updated
+        configuration
+    """
+    for section, sectionconfig in config.items():
+        env_prefix = "SF_{}_".format(section.upper())
+        for key in sectionconfig:
+            env = env_prefix + key.upper()
+            if env in os.environ:
+                config[section][key] = parse_environment_variable(key, os.environ[env])
+
+
+BOOLEAN_KEYS = ("debug", "use_ssl")
+INTEGER_KEYS = ("port")
 
 def parse_environment_variable(key, value):
+    """Parse a value stored in an environment variable.
+
+    Args:
+        key (str): the name of the environment variable
+        value (Union[str, bool, int]): the value obtained from the environment
+            variable
+
+    Returns:
+        [str, bool, int]: the parsed value
+    """
     trues = (True, "true", "True", "TRUE", "1", 1)
     falses = (False, "false", "False", "FALSE", "0", 0)
 
     if key in BOOLEAN_KEYS:
         if value in trues:
             return True
-        elif value in falses:
-            return False
-        else:
-            raise ValueError("Boolean could not be parsed")
-    else:
-        return value
-
-
-class ConfigurationError(Exception):
-    """Exception used for configuration errors"""
 
+        if value in falses:
+            return False
```

Comment:
Much clearer!

Reply:
Was relying on the previous implementation here :)